src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using Raven.Server.ServerWide;
+using Raven.Server.Utils;
+using Sparrow;
+using Sparrow.Global;
+using Sparrow.Utils;
+using Voron;
+using Voron.Platform.Posix;
+
+namespace Raven.Server.Indexing
+{
+    /// <summary>
+    /// This assume single threaded and is meant to be used for indexing only
+    /// </summary>
+    public sealed class TempFileCache : IDisposable
+    {
+        private readonly StorageEnvironmentOptions _options;
+        private readonly Queue<FileStream> _files = new Queue<FileStream>();
+        private readonly Queue<MemoryStream> _ms = new Queue<MemoryStream>();
+
+        private const long MaxFileSizeToReduce = 16 * Constants.Size.Megabyte;
+        private const int MaxFilesInCache = 32;
+
+        public TempFileCache(StorageEnvironmentOptions options)
+        {
+            _options = options;
+            string path = _options.TempPath.FullPath;
+            if (Directory.Exists(path) == false)
+                Directory.CreateDirectory(path);
+            foreach (string file in Directory.GetDirectories(path,  "*" + StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions.TempFileExtension))
```

Comment:
This is the same directory that Voron is using. Can we have a conflict with some file that voron is using?
Also... Voron clears that directory on startup, will we have any files here?

Reply:
That is why I'm using the temp extension.
I'm not _expecting_ to have any files here, but this is there to handle the case of us leaving files around because we crashes, so we'll clean them up.

Reply:
Ok, but can we have a conflict with some of the files that Voron is using? I do not recall, but double-checking.